python/cugraph-dgl/cugraph_dgl/dataloading/dataset.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+from typing import Tuple, Dict
+
+import os
+import cudf
+from cugraph_dgl.dataloading.utils.sampling_helpers import (
+    create_homogeneous_sampled_graphs_from_dataframe,
+    create_heterogeneous_sampled_graphs_from_dataframe,
+)
+
+# TODO: Make optional imports
+import torch
+import dgl
+
+
+# Todo: maybe should switch to __iter__
+class HomogenousBulkSamplerDataset(torch.utils.data.Dataset):
+    def __init__(
+        self,
+        num_batches: int,
+        total_number_of_nodes: int,
+        edge_dir: bool,
+    ):
+        self.num_batches = num_batches
+        self.total_number_of_nodes = total_number_of_nodes
+        self.edge_dir = edge_dir
+        self._current_batch_fn = None
+        self._input_directory = None
+
+    def __len__(self):
+        return self.num_batches
+
+    def __getitem__(self, idx):
+        if self._input_directory is None:
+            raise dgl.DGLError(
+                "Please set input directory by calling `set_input_directory` "
+                "before trying to fetch a sample"
+            )
```

Comment:
So i did not do this because this is set/reset per epoch basis as each epoch has a different directory, Happy to discuss if you have ideas on how to do this cleanly.

Reply:
Let me know if this is fine for now please.